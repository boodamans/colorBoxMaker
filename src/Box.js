import React from 'react';

function Box(props) {
  const { backgroundColor, width, height, onDelete } = props;

  const boxStyle = {
    backgroundColor: backgroundColor || 'lightgray',
    width: width || 100,
    height: height || 100
  };

  return (
    <div style={boxStyle}>
      <button onClick={onDelete}>X</button>
    </div>
  );
}

export default Box;
