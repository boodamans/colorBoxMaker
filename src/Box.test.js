import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

test('Box component renders without crashing', () => {
  render(<Box />);
});

test('Box component matches snapshot', () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

