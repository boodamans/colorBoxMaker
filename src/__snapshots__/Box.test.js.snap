// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: lightgray; width: 100px; height: 100px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
